File: packages/pigeon/platform_tests/alternate_language_test_plugin/example/ios/RunnerTests/NonNullFieldsTest.m

// Copyright 2013 The Flutter Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

@import Flutter;
@import XCTest;

@import alternate_language_test_plugin;

#import "EchoMessenger.h"

///////////////////////////////////////////////////////////////////////////////////////////
@interface NonNullFieldsTest : XCTestCase
@end

///////////////////////////////////////////////////////////////////////////////////////////
@implementation NonNullFieldsTest

- (void)testMake {
  NonNullFieldSearchRequest *request = [NonNullFieldSearchRequest makeWithQuery:@"hello"];
  XCTAssertEqualObjects(@"hello", request.query);
}

- (void)testEquality {
  NonNullFieldSearchRequest *request1 = [NonNullFieldSearchRequest makeWithQuery:@"hello"];
  NonNullFieldSearchRequest *request2 = [NonNullFieldSearchRequest makeWithQuery:@"hello"];
  NonNullFieldSearchRequest *request3 = [NonNullFieldSearchRequest makeWithQuery:@"world"];

  XCTAssertEqualObjects(request1, request2);
  XCTAssertNotEqualObjects(request1, request3);
  XCTAssertEqual(request1.hash, request2.hash);
}

@end

File: packages/pigeon/platform_tests/test_plugin/example/ios/RunnerTests/NonNullFieldsTest.swift
// Copyright 2013 The Flutter Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import Testing

@testable import test_plugin

struct NonNullFieldsTests {
  @Test
  func make() {
    let request = NonNullFieldSearchRequest(query: "hello")
    #expect(request.query == "hello")
  }

  @Test
  func testEquality() {
    let request1 = NonNullFieldSearchRequest(query: "hello")
    let request2 = NonNullFieldSearchRequest(query: "hello")
    let request3 = NonNullFieldSearchRequest(query: "world")

    #expect(request1 == request2)
    #expect(request1 != request3)
    #expect(request1.hashValue == request2.hashValue)
  }
}
